tdms/matlabio/matlabio.h

```diff
@@ -1,19 +1,78 @@
+#pragma once
+#include <algorithm>
 #include <complex>
 #include <string>
 #include "mat_io.h"
 
-double ****castMatlab4DArray(double *array, int nrows, int ncols, int nlayers, int nblocks);
-void freeCastMatlab4DArray(double ****castArray, int nlayers, int nblocks);
-double ***castMatlab3DArray(double *array, int nrows, int ncols, int nlayers);
-void freeCastMatlab3DArray(double ***castArray, int nlayers);
-double **castMatlab2DArray(double *array, int nrows, int ncols);
-void freeCastMatlab2DArray(double **castArray);
-unsigned char ***castMatlab3DArrayUint8(unsigned char *array, int nrows, int ncols, int nlayers);
-void freeCastMatlab3DArrayUint8(unsigned char ***castArray, int nlayers);
-int ***castMatlab3DArrayInt(int *array, int nrows, int ncols, int nlayers);
-void freeCastMatlab3DArrayInt(int ***castArray, int nlayers);
-int **castMatlab2DArrayInt(int *array, int nrows, int ncols);
-void freeCastMatlab2DArrayInt(int **castArray);
+/**
+ * Casts a 4-dimensional array such that it may be indexed according to the usual array indexing
+ * scheme array[l,k,j,i].
+ * @param array is a point to a matlab 4 dimensional array
+ * @param nrows the number of rows in the array
+ * @param ncols the number of columns in the array
+ * @param nlayers the number of layers, each of dimension nrows*ncols
+ * @param nblocks the number of blocks, each of dimension nrows*ncols*nlayers
+ */
+double**** cast_matlab_4D_array(double *array, int nrows, int ncols, int nlayers, int nblocks);
+
+void free_cast_matlab_4D_array(double ****castArray, int nlayers, int nblocks);

```

Comment:
```suggestion
/**
 * Frees the memory of a 4-dimensional array cast using cast_matlab_4D_array
 */
void free_cast_matlab_4D_array(double ****castArray, int nlayers, int nblocks);
```
I _think_ doxygen is smart enough to make this a link to the `cast_` function